### ISSUE TITLE
Support PhET-iO state

### ISSUE THREAD

**Issue by zepumph** [phetsims/projectile-motion#198]
From https://github.com/phetsims/projectile-motion/issues/177,

To support state, the most important items that currently aren't transferred are:

- [x] Trajectories
- [x] modelViewTransformProperty (not instrumented) see https://github.com/phetsims/projectile-motion/issues/207

**Comment by zepumph** [comment]
See also https://github.com/phetsims/projectile-motion/issues/187

**Comment by zepumph** [comment]
I got pretty far on this today. Thanks @samreid for a lot of help.

I created a test case where I loaded studio with one screen, and then fired a single trajectory. After the commits today, the "launched' wrapper looks like this:
![image](https://user-images.githubusercontent.com/6856943/68818739-658b9780-0632-11ea-9adf-be48b371a365.png)

In the state wrapper, I am getting the following error, likely having to do with nested PhetioGroups. Note also that https://github.com/phetsims/phet-io/issues/1570 was an issue I ran into because of nested PhetioGroups.

```
Assertion failed: api mismatches present:

projectileMotion.introScreen.model.trajectoryGroup.trajectory_0.dataPointsGroup.prototype: 
 4. After startup, only dynamic instances prescribed by the baseline file can be registered.
```

Here is a summary of the work thus far:
* Convert DataPoint to es6
* Add a PhetioGroup to Trajectory that handles its data points (called dataPointsGroup). This is on top of the ObservableArray that was already there, which I instrumented with reference serialization. This is because there were a fair number of assumptions about the ordering of the data points, and it felt like I didn't want to rely on PhetioGroup for that, even though in its current implementation. I never saw this fail, but I got the feeling that the orderless state setting could have caused troubles in the array, so keeping a separate, ordered list of the data points seems best. This may be a bit buggy though because don't those numbers get reset on state set? perhaps that is some of the trouble with that extra line in the wrapper picture above.
* Add all fields to constructor that were otherwise just set randomly during runtime and undocumented in the constructor.  This was for DataPoint and Trajectory
* Outfit TrajectoryIO and DataPointIO to hold correct state,

Here is a list of TODOs I kept:
- [ ] undocumented projectileObject.index, likely that needs to be added to the construcor
- [ ] instrument Trajectories.projectileObjects, not sure how yet, another group?!?!
- [ ] Handle rule 4. being broken in the state wrapper
- [ ] instrument rankProperty

**Comment by zepumph** [comment]
As part of Trajectories, we had to instrument two ObservableArrays within Trajectory: projectileObjects and dataPoints.

Originally I tried to have dataPoints be a nested PhetioGroup. This ended up being a total headache. After talking more with @chrisklus and @samreid, we decided that instead I would treat `DataPoint` as a data type serializable. To do this we needed to make sure that DataPoint never shared a reference between multiple Trajectories, which needed to be done, but was a simple improvement. Then things fell into place nicely. 

For the projectile objects I created a formal type instead of what was currently duck typed. This made it easy to make an IO type that could be used for a similar data type serialization strategy. The main hickup here was that each ProjectileObject has a Property. I didn't want to instrument that because then we couldn't use data type serialization for these member of the ObservableArray called `Trajectory.projectileObjects`. I ended up not needing to instrument these, because they are inside of a PhetioGroup, so each time a ProjectileObject is added, the view just listens to that atomic operation. So this sim is currently using the JT strategy instead of the CAF strategy, where the view is able to just listen to the dynamic elements in the model, and we don't need to instrument everything in the view. 

For a second, I thought that we were going to have to instrument each ProjectileObject, and that `projectileObjects` would need to become a PhetioGroup. This was because I didn't quite understand how the view was listening to updates to `ProjectileObject.dataPointProperty`.

I also instrumented the rankProperty. 

For now I'm going to keep this open. I want to think a bit more about the implementation, and perhaps have another look/review next week. For now though I'll move over to https://github.com/phetsims/projectile-motion/issues/207
